src/knex-tables.d.ts

```diff
@@ -0,0 +1,148 @@
+    updated_at: Date;
+    fx_newsletter: boolean;
+    signup_language: string;
+    fxa_refresh_token: null | string;
+    fxa_access_token: null | string;
+    fxa_profile_json: null | unknown;
+    fxa_uid: null | string;
+    // TODO: Find unknown type
+    breaches_last_shown: null | unknown;
+    all_emails_to_primary: boolean;
+    // TODO: Find unknown type
+    breaches_resolved: null | unknown;
+    // TODO: Find unknown type
+    waitlists_joined: null | unknown;
+    // TODO: Find unknown type
+    breach_stats: null | unknown;
+    // TODO: Find unknown type
+    monthly_email_at: null | unknown;
+    // TODO: Find unknown type
+    monthly_email_optout: null | unknown;
+    // TODO: Find unknown type
+    breach_resolution: null | unknown;
+    // TODO: Find unknown type
+    db_migration_1: null | unknown;
+    // TODO: Find unknown type
+    db_migration_2: null | unknown;
+    // TODO: Find unknown type
+    onerep_profile_id: null | unknown;
+    // TODO: Find unknown type
+    email_addresses: unknown[];
+  }
+  type SubscriberOptionalColumns = Extract<
+    keyof SubscriberRow,
+    | "fx_newsletter"
+    | "fxa_access_token"
+    | "fxa_refresh_token"
+    | "fxa_profile_json"
+    | "fxa_uid"
+    | "breaches_last_shown"
+    | "all_emails_to_primary"
+    | "breaches_resolved"
+    | "waitlists_joined"
+    | "breach_stats"
+    | "monthly_email_at"
+    | "monthly_email_optout"
+    | "breach_resolution"
+    | "db_migration_1"
+    | "db_migration_2"
+    | "onerep_profile_id"
+    | "email_addresses"
+  >;
+  type SubscriberAutoInsertedColumns = Extract<
+    keyof SubscriberRow,
+    "id" | "created_at" | "updated_at"
+  >;
+
+  interface EmailAddressRow {
+    id: number;
+    subscriber_id: SubscriberRow["id"];
+    sha1: string;
+    email: string;
+    verification_token: string;
+    verified: boolean;
+    created_at: Date;
+    updated_at: Date;
+  }
+  type EmailAddressAutoInsertedColumns = Extract<
+    keyof EmailAddressRow,
+    "id" | "created_at" | "updated_at"
+  >;
+
+  interface BreachRow {
+    id: number;
+    name: string;
+    title: string;
+    domain: null | string;
+    breach_date: Date;
+    /** Note: added_date is provided by HIBP; this is not the equivalent to created_at in other tables */
+    added_date: Date;
+    /** Note: modified_date is provided by HIBP; this is not the equivalent to updated_at in other tables */
+    modified_date: Date;
+    pwn_count: number;
+    description: null | string;
+    logo_path: string;
+    // TODO: Verify if Knex can actually parse this into a `string[]`
+    data_classes: unknown;
+    is_verified: boolean;
+    is_fabricated: boolean;
+    is_sensitive: boolean;
+    is_retired: boolean;
+    is_spam_list: boolean;
+    is_malware: boolean;
+  }
+  type BreachOptionalColumns = Extract<
+    keyof BreachRow,
+    "domain" | "description"
+  >;
+  type BreachAutoInsertedColumns = Extract<keyof BreachRow, "id">;
+
+  interface Tables {
```

Comment:
They keys in this interface correspond to database tables, and they're used when calling e.g. `knex("subscribers")` to use the correct types for subsequent operations, if possible. See https://knexjs.org/guide/#typescript.